pkgs/development/libraries/science/math/liblapack/default.nix

```
@@ -33,6 +36,26 @@ stdenv.mkDerivation {
 
   doCheck = true;
 
+  # Some CBLAS related tests fail on Darwin:
+  #  14 - CBLAS-xscblat2 (Failed)
+  #  15 - CBLAS-xscblat3 (Failed)
+  #  17 - CBLAS-xdcblat2 (Failed)
+  #  18 - CBLAS-xdcblat3 (Failed)
+  #  20 - CBLAS-xccblat2 (Failed)
+  #  21 - CBLAS-xccblat3 (Failed)
+  #  23 - CBLAS-xzcblat2 (Failed)
+  #  24 - CBLAS-xzcblat3 (Failed)
+  #
+  # Upstream issue to track:
+  # * https://github.com/Reference-LAPACK/lapack/issues/440
+  ctestArgs = lib.optionalString (stdenv.isDarwin && stdenv.isAarch64) "-E '^(CBLAS-(x[sdcz]cblat[23]))$'";
+
+  checkPhase = ''
+    runHook preCheck
+    ctest ${ctestArgs}
```

```suggestion
    ctest $ctestArgs
```
This plays nice with `overrideAttrs` and doesn't require `rec`

---

This doesn't seem to work, with `set -x` I see:

```
running tests
++ runHook preCheck
++ local hookName=preCheck
++ shift
++ local 'hooksSlice=preCheckHooks[@]'
++ local hook
++ for hook in "_callImplicitHook 0 $hookName" ${!hooksSlice+"${!hooksSlice}"}
++ _eval '_callImplicitHook 0 preCheck'
++ declare -F '_callImplicitHook 0 preCheck'
++ eval '_callImplicitHook 0 preCheck'
+++ _callImplicitHook 0 preCheck
+++ local def=0
+++ local hookName=preCheck
+++ declare -F preCheck
+++ type -p preCheck
+++ '[' -n '' ']'
+++ return 0
++ return 0
++ ctest -E
```

---

Right, bash applies both field splitting and filename expansion at the same time when we don't put variable in quotes. We generally want prior but not the latter. A workaround would be to wrap in `set -o noglob`/`set +o noglob`, but we might as well keep this as is. The situation with things like that should improve with advent of `__stucturedAttrs`.